include/fg/defines.h

```diff
@@ -93,10 +93,13 @@ enum ErrorCode {
     FG_ERR_UNKNOWN        = 9003            ///< Unkown error
 };
 
-enum ColorMode {
-    FG_RED =1,                              ///< Single channel
-    FG_RGB =3,                              ///< Three(Red, Green & Blue) channels
-    FG_RGBA=4                               ///< Four(Red, Green, Blue & Alpha) channels
+enum ChannelFormat {
+    FG_GRAY = 100,                          ///< Single channel
```

Comment:
Sorry for repeating this.
But how about using MONO or GRAYSCALE or something on those lines. I'm afraid we might at FG_GRAY to the Color enum at some point.

I guess @pavanky can give his opinion too.